opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -460,16 +470,41 @@ private void formatCardDetails(CardDetails cardDetails) {
         }
     }
 
-    private void startSprayForm(String structureId, String structureUUID, String structureVersion, String structureType,
-                                String taskIdentifier, String taskBusinessStatus, String taskStatus, String propertyType, String sprayStatus, String familyHead) {
-        try {
-            String sprayForm = JsonForm.SPRAY_FORM;
+    private void startForm(Feature feature, CardDetails cardDetails, String encounterType) {
+        String taskBusinessStatus = getPropertyValue(feature, TASK_BUSINESS_STATUS);
+        String taskIdentifier = getPropertyValue(feature, TASK_IDENTIFIER);
+        String taskStatus = getPropertyValue(feature, TASK_STATUS);
+
+        String structureId = feature.id();
+        String structureUUID = getPropertyValue(feature, LOCATION_UUID);
+        String structureVersion = getPropertyValue(feature, LOCATION_VERSION);
+        String structureType = getPropertyValue(feature, LOCATION_TYPE);
+        if (SPRAY_EVENT.equals(encounterType)) {
+            String formString = null;
             if (BuildConfig.BUILD_COUNTRY == Country.NAMIBIA) {
-                sprayForm = JsonForm.SPRAY_FORM_NAMIBIA;
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM_NAMIBIA, listTaskView.getContext());
             } else if (BuildConfig.BUILD_COUNTRY == Country.BOTSWANA) {
-                sprayForm = JsonForm.SPRAY_FORM_BOTSWANA;
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM_BOTSWANA, listTaskView.getContext());
+            } else if (BuildConfig.BUILD_COUNTRY == Country.ZAMBIA) {
+                formString = AssetHandler.readFileFromAssetsFolder(SPRAY_FORM, listTaskView.getContext());
             }
-            String formString = AssetHandler.readFileFromAssetsFolder(sprayForm, listTaskView.getContext());
+            String sprayStatus = cardDetails == null ? null : cardDetails.getSprayStatus();
+            String familyHead = cardDetails == null ? null : cardDetails.getFamilyHead();
+            startForm(formString, structureId, structureUUID, structureVersion, structureType, taskIdentifier,
+                    taskBusinessStatus, taskStatus, structureType, sprayStatus, familyHead);
+        } else if (MOSQUITO_COLLECTION_EVENT.equals(encounterType)) {
+            String formString = null;
+            if (BuildConfig.BUILD_COUNTRY.equals(Country.THAILAND)) {
```

Comment:
For now since we have not implemented the generic form to be used for other countries, comment this so that the Thai is used for all countries and the app does not break.  Tag it with a TODO

Reply:
Done.